engine/src/io/data_parser/DataParser.h

```diff
@@ -30,13 +30,16 @@ class data_parser {
 
 	virtual std::unique_ptr<ral::frame::BlazingTable> parse_batch(
 		std::shared_ptr<arrow::io::RandomAccessFile> file,
-		const std::string & user_readable_file_handle,
 		const Schema & schema,
 		std::vector<size_t> column_indices,
-		size_t row_group) {
+		cudf::size_type row_group) {
 		return nullptr; // TODO cordova ask ALexander why is not a pure virtual function as before
 	}
 
+	virtual size_t get_num_partitions() {
```

Comment:
? num partitions is alwasy 0???

Reply:
if it sa virtual function that the child must implement why not just delete it?

Reply:
The reasoning is, since the reading of a dask-cudf is different from any other kind of input, I needed a function to give me the number of partitions that a dask-cudf contained inside the general algorithm that handles all input types. I agree that this could be improved.

The default value of 0 is to avoid the compiler complains since if I declare the function as abstract, it will force me to implement the function in the other child classes.